src/native/windows/src/handles.c

```diff
@@ -496,13 +496,15 @@ apxCloseHandle(APXHANDLE hObject)
 
     if (IS_INVALID_HANDLE(hObject) || hObject->dwType == APXHANDLE_TYPE_INVALID)
         return FALSE;
-    /* Call the user callback first */
-    (*hObject->fnCallback)(hObject, WM_CLOSE, 0, 0);
-    /* Now go through the callback chain */
-    TAILQ_FOREACH(lpCall, &hObject->lCallbacks, queue) {
+    if (!TAILQ_EMPTY(&hObject->lCallbacks)) {
+        /* Call the user callback first */
+        (*hObject->fnCallback)(hObject, WM_CLOSE, 0, 0);
```

Comment:
Why is this inside the test of `lCallbacks`?

Reply:
This would also throw an access violation, i believe in the JVM.dll which is with a version i do not have pdb files for. I tested this on master with the current 1.2.5 fixes and this could also likely be caused by code from the java side of things. I do not have a good understanding of the issues here but i do know that lCallbacks was empty and that fnCallback would also raise an access violation. What are these meant to be calling? I presume something in the java, however i could not find any documentation, and this is the only "issue" to an otherwise well running service.

Reply:
As per my comment in DAEMON-336, I'd like to see the steps to reproduce this so I can dig into what is going on before applying any patches.

Reply:
@stevehof ping?

Reply:
Only adding the if is probably OK. Something like:
if (lpCall->fnCallback)
   (*lpCall->fnCallback)(hObject, WM_CLOSE, 0, 0);
There are a bunch of place where we test the callback before calling it...